src/main/java/software/amazon/cloudwatchlogs/emf/model/Metadata.java

```diff
@@ -70,6 +70,11 @@ MetricDirective createMetricDirective() {
         return newMetricDirective;
     }
 
+    void setMetricDirective(MetricDirective metricDirective) {
+        cloudWatchMetrics = new ArrayList<>();
```

Comment:
`new ArrayList<>(1)` or `Collections.singletonList(...)` ?

Reply:
As discussed offline, I'll leave this as is for now. For now, we only expect to have one element in this list. But this may change in the future.